### ISSUE TITLE
unexpected token at < JSON

### ISSUE THREAD

**Issue by Samuelyip74** [PatONeil/POS-Terminal#1]
Hi,

I downloaded and populated with sampleDB data.  When I tried to connect to the page, I get this error (as attached).  Can you advise what could be the cause?

Thank.

![error](https://user-images.githubusercontent.com/30211613/47832107-95248480-ddce-11e8-8453-9e0324feb4d0.png)

**Comment by PatONeil** [comment]
It's probably caused by the server not connecting to the MySQL database.    Verify that you have set up the DB correctly and the username and password are correct in sqlconnect script
  
  
  

  
  
  
  
  
>

**Comment by Samuelyip74** [comment]
It shouldn't be error with connect to MySQL.  Because i used createDBs.php script to create the MySQL database tables.  If the username and password is wrong, I shouldn't be able to create the tables.

Any idea?

**Comment by tmhevencent** [comment]
Was this issue solved?

**Comment by PatONeil** [comment]
Not sure, was problem reported.    When/where does it happen.
  
  
  

  
  
  
  
  
>

**Comment by tmhevencent** [comment]
> Hi,
> 
> I downloaded and populated with sampleDB data. When I tried to connect to the page, I get this error (as attached). Can you advise what could be the cause?
> 
> Thank.
> 
> ![error](https://user-images.githubusercontent.com/30211613/47832107-95248480-ddce-11e8-8453-9e0324feb4d0.png)

This issue, i am also getting it but by disabling die() inside connectPOS, myErrorHandler function it works ok

**Comment by tmhevencent** [comment]
> Not sure, was problem reported. When/where does it happen.
> […](#)
> On Dec 6, 2018 at 2:34 AM, <tmhevencent> wrote: Was this issue solved? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

when u access the url to display the menu 
![image](https://user-images.githubusercontent.com/5189237/49585739-455a4f80-f970-11e8-8e3b-f1e81dbf66f2.png)

**Comment by PatONeil** [comment]
The probàble cause is thefailure to connect to MySQL database.  Check the DB name you used, username and password.in connect script..

**Comment by tmhevencent** [comment]
I doubt, once I comment php die function everything is OK.

**Comment by FlorianLu** [comment]
THIS IS INTENTIONALLY BADLY MADE SO THAT YOU ASK QUESTIONS !LAME!!

**Comment by Eleberria** [comment]
This thread is a bit old but I hope someone can find this helpful. First, excuse my English, I'll try my best.
 I've got the same error when, once installed and  populated the database with sampleDB data, I loaded the site in the browser. I've been checking the source code and in the configuration file, "connectPOS.php", I've found these lines:
$hDir = "../log";
error_reporting(E_ALL);
ini_set('error_log',$hDir."/PHP_Errors.txt");
Basically, when you first load it, the system tries to create a log file in a "log" folder located in the root directory, but I've checked the directory and it didn't contain a "log" folder. So, I created the folder, reloaded the page and, voila, everything worked as expected.

**Comment by Eleberria** [comment]
My recommendation, edit "connectPOS.php" adding these lines:

$hDir = "../log";
if (!file_exists($hDir)) {
    mkdir($hDir, 0777, true);
}
error_reporting(E_ALL);
ini_set('error_log',$hDir."/PHP_Errors.txt");

You can edit permissions as you want, 0777 is by default

**Comment by Eleberria** [comment]
By the way.... I almost forgot... Happy Saint Patrick's Day!!!!! ;-)

**Comment by kingerypro** [comment]
Upon updating everything, ive ran into one last issue, when i go to complete the order it throws back:

posTerminal.posMenuOrderMgr:parserror SyntaxError: JSON Parse error: Unrecognized token'<'

I am aware of the previous pages needing to be updated and the folder created for your log but i am confused, ive searched through the code and i cant seem to even see the direction of the issue.

**Comment by kingerypro** [comment]
Upon checking with the print ability, the system comes back with:
Print Order Mgr: error
as well as editing the order information is throwing back nothing or empty orders after clicking edit on specific orders.

Ive searched through the print order mgr js file and cant really see the issue, thanks for any help in advance.

**Comment by kingerypro** [comment]
> This thread is a bit old but I hope someone can find this helpful. First, excuse my English, I'll try my best.
> I've got the same error when, once installed and populated the database with sampleDB data, I loaded the site in the browser. I've been checking the source code and in the configuration file, "connectPOS.php", I've found these lines:
> $hDir = "../log";
> error_reporting(E_ALL);
> ini_set('error_log',$hDir."/PHP_Errors.txt");
> Basically, when you first load it, the system tries to create a log file in a "log" folder located in the root directory, but I've checked the directory and it didn't contain a "log" folder. So, I created the folder, reloaded the page and, voila, everything worked as expected.

Def helped me allot, thanks! :)
